etna/datasets/tsdataset.py

```diff
@@ -1210,29 +1213,36 @@ def train_test_split(
         if train_start_defined < self.df.index.min():
             warnings.warn(f"Min timestamp in df is {self.df.index.min()}.")
 
-        train_df = self.df.loc[train_start_defined:train_end_defined][self.raw_df.columns]  # type: ignore
+        # TODO: why do we use self.raw_df.columns instead of self.df.columns? Need to be discussed
+        train_df_init = self.df.loc[train_start_defined:train_end_defined][self.raw_df.columns]  # type: ignore
+        train_df = self.df.loc[train_start_defined:train_end_defined][self.df.columns]  # type: ignore
         train_raw_df = self.raw_df.loc[train_start_defined:train_end_defined]  # type: ignore
         train = TSDataset(
```

Comment:
It may be much easier to make a copy of `self` and make `iloc` on `df` and `raw_df` of this copy.

Reply:
We could also try to avoid copying `df_exog`.